generators/php/lists.js

```diff
@@ -0,0 +1,348 @@
+  return [code, Blockly.PHP.ORDER_FUNCTION_CALL];
+};
+
+Blockly.PHP['lists_getIndex'] = function(block) {
+  // Get element at index.
+  var mode = block.getFieldValue('MODE') || 'GET';
+  var where = block.getFieldValue('WHERE') || 'FROM_START';
+  var at = Blockly.PHP.valueToCode(block, 'AT',
+      Blockly.PHP.ORDER_UNARY_NEGATION) || '1';
+  var list = Blockly.PHP.valueToCode(block, 'VALUE',
+      Blockly.PHP.ORDER_FUNCTION_CALL) || 'array()';
+
+  if (where == 'FIRST') {
+    if (mode == 'GET') {
+      var code = list + '[0]';
+      return [code, Blockly.PHP.ORDER_FUNCTION_CALL];
+    } else if (mode == 'GET_REMOVE') {
+      var code = 'array_shift(' + list + ')';
+      return [code, Blockly.PHP.ORDER_FUNCTION_CALL];
+    } else if (mode == 'REMOVE') {
+      return 'array_shift(' + list + ');\n';
+    }
+  } else if (where == 'LAST') {
+    if (mode == 'GET') {
+      var code = 'end(' + list + ')';
+      return [code, Blockly.PHP.ORDER_FUNCTION_CALL];
+    } else if (mode == 'GET_REMOVE') {
+      var code = 'array_pop(' + list + ')';
+      return [code, Blockly.PHP.ORDER_FUNCTION_CALL];
+    } else if (mode == 'REMOVE') {
+      return 'array_pop(' + list + ');\n';
+    }
+  } else if (where == 'FROM_START') {
+    // Blockly uses one-based indicies.
+    if (Blockly.isNumber(at)) {
+      // If the index is a naked number, decrement it right now.
+      at = parseFloat(at) - 1;
+    } else {
+      // If the index is dynamic, decrement it in code.
+      at += ' - 1';
+    }
+    if (mode == 'GET') {
+      var code = list + '[' + at + ']';
+      return [code, Blockly.PHP.ORDER_FUNCTION_CALL];
+    } else if (mode == 'GET_REMOVE') {
+      var code = 'array_splice(' + list + ', ' + at + ', 1)[0]';
+      return [code, Blockly.PHP.ORDER_FUNCTION_CALL];
+    } else if (mode == 'REMOVE') {
+      return 'array_splice(' + list + ', ' + at + ', 1);\n';
+    }
+  } else if (where == 'FROM_END') {
+    if (mode == 'GET') {
+      var code = 'array_slice(' + list + ', -' + at + ', 1)[0]';
+      return [code, Blockly.PHP.ORDER_FUNCTION_CALL];
+    } else if (mode == 'GET_REMOVE' || mode == 'REMOVE') {
+      code = 'array_splice(' + list + ', count(' + list + ') - ' + at + ', 1)[0]';
+      if (mode == 'GET_REMOVE') {
+        return [code, Blockly.PHP.ORDER_FUNCTION_CALL];
+      } else if (mode == 'REMOVE') {
+        return code + ';\n';
+      }
+    }
+  } else if (where == 'RANDOM') {
+    if (mode == 'GET'){
+      code = list + '[rand(0,count(' + list + ')-1)]';
+      return [code, Blockly.PHP.ORDER_FUNCTION_CALL];
+    } else if (mode == 'GET_REMOVE'){
+      var functionName = Blockly.PHP.provideFunction_(
+          'lists_get_remove_random_item',
+          [ 'function ' + Blockly.PHP.FUNCTION_NAME_PLACEHOLDER_ +
+          '() {',
+            '  global ' + list + ';',
+            '  $x = rand(0,count($list)-1);',
+            '  unset($list[$x]);',
+            '  return array_values($list);',
```

Comment:
If the function returns the removed element, the test fails. Is the function or the test wrong?